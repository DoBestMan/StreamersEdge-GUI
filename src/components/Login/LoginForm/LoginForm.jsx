/**
 * Form that handles account creation
 */
import React, {Component} from 'react';
import FormControl from '@material-ui/core/FormControl';
import SignupInput from '../../SignupInput';
import IconPassword from '../../../assets/images/login/Password.png';
import IconPasswordActive from '../../../assets/images/login/Password_Over.png';
import IconUsername from '../../../assets/images/login/Username_1x.png';
import IconUsernameActive from '../../../assets/images/login/Username_Over.png';
import {Link} from 'react-router-dom';
import Button from '@material-ui/core/Button';
import LoginButton from '../../../assets/images/login/login_button.png';
import LoginButtonActive from '../../../assets/images/login/login_active_button.png';
import AuthService from '../../../services/AuthService';
import ProfileService from '../../../services/ProfileService';
import AuthFooter from '../../Auth/AuthFooter';
import LogoImage from '../../../assets/images/se-logo-stacked.png';
import {translate} from '../../../utility/GeneralUtils';

class LoginForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      username: '',
      password: ''
    };
  }

  handleSubmit = (event) => {
    event.preventDefault();

    if (this.state.username < 3 || this.state.password < 4) {
      return;
    }

    const account = {
      login: this.state.username,
      password: this.state.password
    };

    this.props.handleLogin(account);
  };

  handleChange = (event) => {
    const {name, value} = event.target;
    this.setState({
      [name]: value
    });
  };

  logout = () => {
    AuthService.logout();
  };

  getProfile = () => {
    ProfileService.getProfile();
  };

  render() {
    return (
      <>
        <form className='login-form' onSubmit={ this.handleSubmit }>
          <img src={ LogoImage } alt='logo' />
          <FormControl margin='none' required>
            <SignupInput
              name='username'
              handleChange={ this.handleChange }
              inputValue={ this.state.username }
              inputImage={ IconUsername }
              activeInputImage={ IconUsernameActive }
              placeholder={ translate('login.enterUsername') }
            />
          </FormControl>
          <FormControl margin='none' required>
            <SignupInput
              name='password'
              type='password'
              handleChange={ this.handleChange }
              inputValue={ this.state.password }
              inputImage={ IconPassword }
              activeInputImage={ IconPasswordActive }
              placeholder={ translate('login.enterPassword') }
            />
          </FormControl>
          {this.props.errorText}
          <span className='register__textlink'>
            {translate('login.dontHaveAccount')}
            <span onClick = { this.props.goRegister } className='goregister__link'>{translate('login.register')}</span>
          </span>
          <span className='register__textlink'>
            <Link className='login__link' to={ '/forgot-password' } activeclassname='active'>
              {translate('login.forgotPass')}
            </Link>
          </span>
          <div className='login__btn-container'>
            <Button disabled={ false } className='login__btn' type='submit' style={ {color: 'white'} }>
              <img
                className='login__btn-img'
                src={ LoginButton }
                alt='Login'
                type='submit'
                onMouseOver={ (e) => (e.currentTarget.src = LoginButtonActive) }
                onMouseOut={ (e) => (e.currentTarget.src = LoginButton) }
              />
            </Button>
          </div>
        </form>
        <AuthFooter />
      </>
    );
  }
}

export default LoginForm;
