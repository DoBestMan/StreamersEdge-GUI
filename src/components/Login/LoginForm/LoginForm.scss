@import '../../../assets/styles/common.scss';

.login {
  &-form {
    @extend .txt--white;

    display: flex;
    flex-direction: column;
    align-items: center;

    &__input {
      @extend .txt--white;

      margin-top: 6px;
      margin-bottom: 6px;
    }
  }

  &__btn {
    &-container {
      @extend .txt--white;

      position: relative;
      padding-top: 25px;
      text-align: center;
    }

    &-txt {
      position: absolute;
      top: 50%;
      left: 50%;
      transform: translate(-50%, -50%);
    }
  }

  &__link {
    @extend .txt--white;
  }
}

.goregister__link {
  @extend .txt--white;

  padding-left: 5px;
  text-decoration: underline;
  cursor: pointer;
}

.register {
  &__text-link {
    display: flex;
    justify-content: center;
    margin-top: 25px;
    margin-bottom: 25px;
  }
}
