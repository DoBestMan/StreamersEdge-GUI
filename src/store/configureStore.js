import {applyMiddleware, createStore} from 'redux';
import {createHashHistory} from 'history';
import {routerMiddleware} from 'connected-react-router';
import Immutable from 'immutable';
import {composeWithDevTools} from 'redux-devtools-extension';
import createRootReducer from '../reducers/index';
import thunk from 'redux-thunk';

// Session history
export const history = createHashHistory();

export default function configureStore() {
  // For typescript, this may need tweaking...
  let initialState = Immutable.Map();

  // Define var for filtering actions when debugging redux to reduce clutter and increase performance.
  let actionsWhitelist = [];

  // Configure enhancer for redux dev tools extensions (if available)
  const composeEnhancers = composeWithDevTools({
    features: {
      dispatch: true
    },
    // Option for immutable
    serialize: {immutable: Immutable},
    actionsWhitelist: actionsWhitelist
  });

  // Construct enhancer
  const enhancer = composeEnhancers(
    applyMiddleware(thunk, routerMiddleware(history))
  );
  const store = createStore(
    createRootReducer(history),
    initialState,
    enhancer
  );

  return store;
}