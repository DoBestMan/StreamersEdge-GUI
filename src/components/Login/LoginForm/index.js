import LoginForm from './LoginForm';
import './LoginForm.scss';

export default LoginForm;