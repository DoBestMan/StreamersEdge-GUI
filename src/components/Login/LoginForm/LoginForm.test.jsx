import * as React from 'react';
import {Provider} from 'react-redux';
import configureStore from '../../../store/configureStore';
import {shallow} from 'enzyme';
import LoginForm from './LoginForm';

describe('<LoginForm/>', () => {
  const store = configureStore();
  test('should render correctly', () => {
    const component = shallow(
      <Provider store={ store }>
        <LoginForm />
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
});