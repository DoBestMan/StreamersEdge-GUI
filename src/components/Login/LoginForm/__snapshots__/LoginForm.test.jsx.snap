// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm/> should render correctly 1`] = `ShallowWrapper {}`;
